engine/xml-config/src/main/java/org/datacleaner/job/JaxbJobReader.java

```diff
@@ -947,17 +948,26 @@ private void applyProperties(final ComponentBuilder builder,
                     String stringValue = getValue(property);
                     if (stringValue == null) {
                         final String variableRef = property.getRef();
+
                         if (variableRef == null) {
-                            throw new IllegalStateException(
-                                    "Neither value nor ref was specified for property: " + name);
+                            String templateValue = property.getTemplate();
+                            if (templateValue != null) {
+                                templateValue = templateValue.replace("${","");
+                                for (final Entry<String, String> variable : variables.entrySet()) {
+                                    templateValue = templateValue.replace(variable.getKey(), variable.getValue());
+                                }
+                                templateValue = templateValue.replace("}","");
```

Comment:
I'm not entirely sure what's going on here. It should only look for the value key inside `${}` and only individually.

`${my.property.key.a} + ${my.property.key.b}` should match. `${my.property.key.a + my.property.key.b}` should not.

We need to be exact here.